staging/src/k8s.io/component-base/kflag/kflag.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kflag
+
+import (
+	"github.com/spf13/pflag"
+	utilflag "k8s.io/apiserver/pkg/util/flag"
+)
+
+type FlagSet struct {
+	fs *pflag.FlagSet
+}
+
+// For now, from a migration standpoint, it probably makes sense to re-use existing pflag flagsets,
+// since components may register flags against them in multiple places and need to incrementally
+// migrate, or do other custom things like set the normalize func. In the future, we should
+// probably be constructing the internal pflag flagset on behalf of the user, with a common
+// normalize func, etc.
+func NewFlagSet(fs *pflag.FlagSet) *FlagSet {
+	return &FlagSet{
+		fs: fs,
+	}
+}
+
+func (fs *FlagSet) Parse(args []string) {
+	fs.fs.Parse(args)
+}
+
+// Basic Int32Var approach, including a helper so components don't
+// have to repeatedly implement the basic apply for int32s.
+
+type Int32Value struct {
+	name  string
+	value int32
+	fs    *pflag.FlagSet
+}
+
+func (fs *FlagSet) Int32Var(name string, def int32, usage string) *Int32Value {
+	v := &Int32Value{
+		name: name,
+		fs:   fs.fs,
+	}
+	fs.fs.Int32Var(&v.value, name, def, usage)
+	return v
+}
+
+// Replace the int32 altogether
+func (v *Int32Value) Replace(target *int32) {
```

Comment:
`Replace` is a strange name. `ApplyIfChanged` ?

Reply:
I thought about just calling this `Set`, but given that there is a `Replace` vs. `Merge` distinction for other types, and `Replace` and `Set` basically mean the same thing, I figured we could keep a consistent naming convention and call it `Replace` for all types. Alternatively, we could just call it `Set` for all types, which might be clearer (as you mentioned below).

I'd like to leave off the `IfChanged`, since the plan is for all apply methods to be conditional on the flag being set.